### PR TITLE
[KYUUBI #5756] Introduce specified initialized SQL  to every engine

### LINKED ISSUE #5756: [TASK][EASY] Introduce specified initialized SQL to every engine
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

As of nowadays,  the configuration key `kyuubi.engine.session.initialize.sql` is used in both Flink and Spark engines.  To specify the initialized SQL for each engine, we need to introduce a separate key.

For example:
`kyuubi.engine.flink.initialize.sql`
`kyuubi.session.engine.flink.initialize.sql`

`kyuubi.engine.spark.initialize.sql`
`kyuubi.session.engine.spark.initialize.sql`

### How should we improve?

Add new ConfigEntries to Flink and Spark engines

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by hadoopkandy**
# :mag: Description
## Issue References 🔗

This pull request fixes #5756

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [x] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public KyuubiBeeLine(boolean isBeeLine) {
   @Override
   void usage() {
     super.usage();
-    output("Usage: java \" + KyuubiBeeLine.class.getCanonicalName()");
+    output("Usage: java " + KyuubiBeeLine.class.getCanonicalName());
```
Comment (pan3793) [inline kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java:88]: this is an independent issue, you can fix a in an independent PR first :)

**Comment by pan3793** [comment]
cc @YesOrNo828

File: kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java (hunk 1 above)
Comment (hadoopkandy) [reply kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiBeeLine.java]: ok

File: docs/configuration/settings.md [hunk 2]
```diff
@@ -142,6 +142,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.engine.flink.initialize.sql                       |                           | the initialize sql for flink engine. it fallback to  `kyuubi.engine.initialize.sql`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | seq      | 1.8.1 |
```
Comment (YesOrNo828) [inline docs/configuration/settings.md:145]:
```suggestion
| kyuubi.engine.flink.initialize.sql                       |                           | The initialize sql for flink engine. It fallback to  `kyuubi.engine.initialize.sql`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | seq      | 1.8.1 |
```

File: docs/configuration/settings.md [hunk 3]
```diff
@@ -177,6 +178,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
+| kyuubi.engine.spark.initialize.sql                       |                           | the initialize sql for spark engine. it it fallback to  `kyuubi.engine.session.initialize.sql`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | seq      | 1.8.1 |
```
Comment (YesOrNo828) [inline docs/configuration/settings.md:181]:
```suggestion
| kyuubi.engine.spark.initialize.sql                       |                           | The initialize sql for spark engine. It fallback to  `kyuubi.engine.initialize.sql`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | seq      | 1.8.1 |
```

File: docs/configuration/settings.md [hunk 4]
```diff
@@ -429,6 +431,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
 | kyuubi.session.engine.flink.max.rows                 | 1000000                 | Max rows of Flink query results. For batch queries, rows exceeding the limit would be ignored. For streaming queries, the query would be canceled if the limit is reached.                                                                                                                                                                                                                                               | int      | 1.5.0 |
+| kyuubi.session.engine.flink.initialize.sql           |                         | the initialize sql for flink session.it fallback to  `kyuubi.engine.session.initialize.sql`.                                                                                                                                                                                                                                                                                                                             | seq      | 1.8.1 |
```
Comment (YesOrNo828) [inline docs/configuration/settings.md:434]:
```suggestion
| kyuubi.session.engine.flink.initialize.sql           |                         | The initialize sql for Flink session. It fallback to `kyuubi.engine.session.initialize.sql`.                                                                                                                                                                                                                                                                                                                             | seq      | 1.8.1 |
```

File: docs/configuration/settings.md [hunk 5]
```diff
@@ -444,6 +447,7 @@ You can configure the Kyuubi properties in `$KYUUBI_HOME/conf/kyuubi-defaults.co
...
 | kyuubi.session.engine.spark.showProgress             | false                   | When true, show the progress bar in the Spark's engine log.                                                                                                                                                                                                                                                                                                                                                              | boolean  | 1.6.0 |
+| kyuubi.session.engine.spark.initialize.sql           |                         | the initialize sql for spark session. it fallback to  `kyuubi.engine.session.initialize.sql`                                                                                                                                                                                                                                                                                                                             | seq      | 1.8.1 |
```
Comment (YesOrNo828) [inline docs/configuration/settings.md:450]:
```suggestion
| kyuubi.session.engine.spark.initialize.sql           |                         | The initialize sql for spark session. It fallback to `kyuubi.engine.session.initialize.sql`.                                                                                                                                                                                                                                                                                                                             | seq      | 1.8.1 |
```

**Review by YesOrNo828 — COMMENTED** [review]
@hadoopkandy  Thanks, I left some minor comments, please take a look.

File: docs/configuration/settings.md (hunk 2 above)
Comment (wForget) [reply docs/configuration/settings.md:145]: The setting file is regenerated using the script, we should change the description of conf.

FYI: https://kyuubi.readthedocs.io/en/master/contributing/code/developer.html#append-descriptions-of-new-configurations-to-settings-md

**Review by YesOrNo828 — APPROVED** [review]
LGTM

File: docs/configuration/settings.md (hunk 2 above)
Comment (hadoopkandy) [reply docs/configuration/settings.md]: Thanks, After adopting your suggestion, it passed the related CI test, otherwise it won't be able to pass

**Comment by hadoopkandy** [comment]
> @hadoopkandy Thanks, I left some minor comments, please take a look.

Thanks, I committed it again

**Review by pan3793 — APPROVED** [review]

**Pull request closed without merging**